index.js

```diff
@@ -41,87 +43,79 @@ const displayGuestScore = (score) => (guestPoint.textContent = score);
   let rVal = roundVal;
-  if (rVal == 1){
+  if (rVal === 1){
     homeRound1Score.textContent = hScore;
     guestRound1Score.textContent = gScore;
   }
-  if (rVal == 2){
+  if (rVal === 2){
     homeRound2Score.textContent = hScore;
     guestRound2Score.textContent = gScore;
   }
-  if (rVal == 3){
+  if (rVal === 3){
     homeRound3Score.textContent = hScore;
     guestRound3Score.textContent = gScore;
   }
-  if (rVal == 4){
+  if (rVal === 4){
     homeRound4Score.textContent = hScore;
     guestRound4Score.textContent = gScore;
     calculateFinalScore();
-  }  
+  } 
 }
 
 function calculateFinalScore() {
   hFinalScore = parseInt(homeRound1Score.textContent) + parseInt(homeRound2Score.textContent) + parseInt(homeRound3Score.textContent) + parseInt(homeRound4Score.textContent);
   gFinalScore = parseInt(guestRound1Score.textContent) + parseInt(guestRound2Score.textContent) + parseInt(guestRound3Score.textContent) + parseInt(guestRound4Score.textContent);
+  
   totalHomeScore.textContent = hFinalScore;
   totalGuestScore.textContent = gFinalScore;
-  if (hFinalScore > gFinalScore) {
-    finalMsg.textContent = "The Winner is Team Home 🥳";
-  }
-  else if (hFinalScore < gFinalScore) {
-    finalMsg.textContent = "The Winner is Team Guest 🥳";
-  }
-  else {
-    finalMsg.textContent = "The Match is a Draw!";
-  }
+
+  const finalMessage = hFinalScore > gFinalScore ? "The Winner is Team Home 🥳"
+    : hFinalScore < gFinalScore ? "The Winner is Team Guest 🥳"
+    : "The Match is a Draw!"
+  
+  finalMsg.textContent = finalMessage;
 }
 
 addPointButton.forEach((btn) => {
   btn.addEventListener("click", function (e) {
     const buttonIsWithAttribute = e.target.getAttribute("data-addPointAttribute");
-  
-    if (buttonIsWithAttribute === "home-add-1") {
-      addHomeTotalScore(1);
-      displayHomeScore(homeTotalScore);
-    }
-  
-    if (buttonIsWithAttribute === "home-add-2") {
-      addHomeTotalScore(2);
-      displayHomeScore(homeTotalScore);
-    }
-  
-    if (buttonIsWithAttribute === "home-add-3") {
-      addHomeTotalScore(3);
-      displayHomeScore(homeTotalScore);
-    }
-  
-    if (buttonIsWithAttribute === "guest-add-1") {
-      addGuestTotalScore(1);
-      displayGuestScore(guestTotalScore);
-    }
-  
-    if (buttonIsWithAttribute === "guest-add-2") {
-      addGuestTotalScore(2);
-      displayGuestScore(guestTotalScore);
-    }
-  
-    if (buttonIsWithAttribute === "guest-add-3") {
-      addGuestTotalScore(3);
-      displayGuestScore(guestTotalScore);
-    }   
+
+    const hNum = buttonIsWithAttribute === "home-add-1" ? 1
+      : buttonIsWithAttribute === "home-add-2" ? 2
+      : buttonIsWithAttribute === "home-add-3" ? 3 : 0
+    
+    addHomeTotalScore(hNum);
+    displayHomeScore(homeTotalScore);
+
+    const gNum = buttonIsWithAttribute === "guest-add-1" ? 1
+      : buttonIsWithAttribute === "guest-add-2" ? 2
+      : buttonIsWithAttribute === "guest-add-3" ? 3 : 0
```

Comment:
- [ ] Here, still the value is being checked one by one. If we have 10 items here there will be 10 condition checks which are so expensive. So make sure to use a common logic that works for any number and both home and guest.

Reply:
Resolved